File: src/main/java/me/mixces/animatium/util/EntityUtils.java
package me.mixces.animatium.util;

import net.minecraft.client.render.entity.state.EntityRenderState;
import net.minecraft.entity.Entity;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public abstract class EntityUtils {
    private static final Map<EntityRenderState, Entity> STATE_TO_ENTITY = new HashMap<>();

    public static void setEntityByState(EntityRenderState state, Entity entity) {
        STATE_TO_ENTITY.put(state, entity);
    }

    public static Optional<Entity> getEntityByState(EntityRenderState state) {
        return Optional.ofNullable(STATE_TO_ENTITY.getOrDefault(state, null));
    }
}

File: src/main/kotlin/me/mixces/animatium/util/EntityUtils.kt
package me.mixces.animatium.util

import net.minecraft.client.render.entity.state.EntityRenderState
import net.minecraft.entity.Entity
import java.util.Optional

abstract class EntityUtils {
    companion object {
        val STATE_TO_ENTITY = hashMapOf<EntityRenderState, Entity>()

        @JvmStatic
        fun setEntityByState(state: EntityRenderState, entity: Entity) {
            STATE_TO_ENTITY.put(state, entity)
        }

        @JvmStatic
        fun getEntityByState(state: EntityRenderState): Optional<Entity> {
            return Optional.ofNullable(STATE_TO_ENTITY.getOrDefault(state, null))
        }
    }
}